text/0001-agile-coretime.md

```diff
@@ -0,0 +1,652 @@
+
+#### Intra-Leadin Price-decrease
+
+During the Leadin Period of a sale, the effective price starts higher than the Sale Price and falls to end at the Sale Price at the end of the Leadin Period. The price can thus be defined as a simple factor above one on which the Sale Price is multiplied. A function which returns this factor would accept a factor between zero and one specifying the portion of the Leadin Period which has passed.
+
+Thus we assume `SALE_PRICE`, then we can define `PRICE` as:
+
+```
+PRICE := SALE_PRICE * FACTOR((NOW - LEADIN_BEGIN) / LEADIN_PERIOD)
+```
+
+We can define a very simple progression where the price decreases monotonically from double the Sale Price at the beginning of the Leadin Period.
+
+```
+FACTOR(T) := 2 - T
+```
+
+#### Parameter Values
+
+Parameters are either *suggested* or *specified*. If *suggested*, it is non-binding and the proposal should not be judged on the value since other RFCs and/or the governance mechanism of Polkadot is expected to specify/maintain it. If *specified*, then the proposal should be judged on the merit of the value as-is.
+
+| Name                | Value                        | |
+| ------------------- | ---------------------------- | ---------- |
+| `BULK_PERIOD`       | `28 * DAYS`                  | specified  |
+| `INTERLUDE_PERIOD`  | `7 * DAYS`                   | specified  |
+| `LEADIN_PERIOD`     | `7 * DAYS`                   | specified  |
+| `TIMESLICE`         | `8 * MINUTES`                | specified  |
+| `BULK_TARGET`       | `30`                         | suggested  |
+| `BULK_LIMIT`        | `45`                         | suggested  |
+| `RENEWAL_PRICE_CAP` | `Perbill::from_percent(2)`   | suggested  |
+
+
+#### Instantaneous Price Progression
+
+This proposal assumes the existence of a Relay-chain-based price-setting mechanism for the Instantaneous Coretime Market which alters from block to block, taking into account several parameters: the last price, the size of the Instantaneous Coretime Pool (in terms of cores per Relay-chain block) and the amount of Instantaneous Coretime waiting for processing (in terms of Core-blocks queued).
+
+The ideal situation is to have the size of the Instantaneous Coretime Pool be equal to some factor of the Instantaneous Coretime waiting. This allows all Instantaneous Coretime sales to be processed with some limited latency while giving limited flexibility over ordering to the Relay-chain apparatus which is needed for efficient operation.
+
+If we set a factor of three, and thus aim to retain a queue of Instantaneous Coretime Sales which can be processed within three Relay-chain blocks, then we would increase the price if the queue goes above three times the amount of cores available, and decrease if it goes under.
+
+Let us assume the values `OLD_PRICE`, `FACTOR`, `QUEUE_SIZE` and `POOL_SIZE`. A simple definition of the `NEW_PRICE` would be thus:
+
+```
+NEW_PRICE := IF QUEUE_SIZE < POOL_SIZE * FACTOR THEN
```

Comment:
I think the very least we would need to use some average value of `POOL_SIZE`, given that the number of on-demand cores can essentially change very block, given for example that regions with only a single bit set can be assigned to the instantaneous market.

Reply:
It's not intended to be anything other than an illustration.